### PR TITLE
Task 7

### LINKED ISSUE #21: Improve authorization checks
Replace `current_user.admin?` checks with `can? :admin, User` as it is showed in CanCan docs https://github.com/ryanb/cancan/wiki/checking-abilities#checking-with-class

### PR DESCRIPTION
**Pull request by Skysplit**
This PR is based off on branch `task-6`. It should be rebased and merged to master after changes to `task-6` branch are made.

This also closes #21

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -6,7 +6,17 @@ class UsersController < ApplicationController
...
-    @users = User.includes(:interests).all
+    @users = User.listing.includes(:interests).all
+
+    respond_to do |format|
+      format.html
+      format.csv do
+        authorize! :admin, User
+        headers = ['id', 'email', 'gender', 'age']
+        values = @users.map { |user| user.slice(*headers).values }
+        send_data CsvHelper.create_csv_string(headers: headers, values: values), filename: 'users.csv', type: 'text/csv'
```
Comment (socjopata) [inline app/controllers/users_controller.rb]: Consider formatting a code in a way, that the horizontal scrollbar doesn't appear. It's easier to read the code this way ;)

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -16,7 +26,8 @@ def destroy
   end
 
   private
+
   def get_user
-    @user = User.find(params[:id])
+    @user = User.find params[:id]
```
Comment (socjopata) [inline app/controllers/users_controller.rb]: I understand that parenthesis is optional in Ruby, however I am a big fan of using it, except a few cases where we are dealing with DSL like DSL provided for instance by rspec. Here is a nice, free screencast, which would give you some backround, why I have a strong opinion on the matter (https://www.rubytapas.com/2012/11/16/episode-024-incidental-change/). Let me know if you would like to discuss that ;)

File: spec/helpers/csv_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      values = [["a", "b", "c"]]
+
+      expect(CsvHelper.create_csv_string(values: values)).to eq (values.first.join(',') + "\n")
+    end
+    
+    it "should return csv with headers and values" do
+      headers = ['test', 'test']
+      values = [['lorem', 'ipsum']]
+      
+      expected = headers.map(&:humanize).join(',') + "\n"
```
Comment (socjopata) [inline spec/helpers/csv_helper_spec.rb]: It looks like you're partially copying the implementation of the very method you want to test. It's a bit like repeating yourself. There is nothing wrong with defining expectations explicitly, as strings ;)

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (Skysplit) [inline app/controllers/users_controller.rb]: @socjopata I think I've fallen in love with skipping parenthesis too much :) This is actually a really good reason to use parenthesis (although I think code without it looks way more awesome!).  
Can't really argue with that.

I'll keep that in mind for the future. Thank you!

**Pull request merged**